benchmarks/sort/sort_benchmark.cpp

```diff
@@ -0,0 +1,661 @@
+    Kokkos::parallel_reduce(
+        "find_min_max_view", Kokkos::RangePolicy<ExecutionSpace>(0, n),
+        Kokkos::Impl::min_max_functor<ViewType>(view), reducer);
+
+    Kokkos::BinSort<ViewType, CompType, typename ViewType::device_type,
+                    SizeType>
+        bin_sort(view, CompType(n / 2, result.min_val, result.max_val), true);
+    bin_sort.create_permute_vector();
+    bin_sort.sort(view);
+
+    return bin_sort.get_permute_vector();
+  }
+
+  static void sort(Kokkos::View<ValueType *, DeviceType> view)
+  {
+    auto permute = sortAndComputePermutation(view);
+    std::ignore = permute;
+  }
+
+  static void applyPermutation(Kokkos::View<SizeType *, DeviceType> permute,
+                               Kokkos::View<ValueType *, DeviceType> in,
+                               Kokkos::View<ValueType *, DeviceType> &out)
+  {
+    using ExecutionSpace = typename DeviceType::execution_space;
+    int const n = in.extent(0);
+    Kokkos::parallel_for(
+        Kokkos::RangePolicy<ExecutionSpace>(0, n),
+        KOKKOS_LAMBDA(int const i) { out(permute(i)) = in(i); });
+  }
+};
+
+#if defined(KOKKOS_ENABLE_SERIAL)
+template <typename ValueType, typename DeviceType, typename SizeType>
+struct SortStd
+{
+  using value_type = ValueType;
+  using device_type = DeviceType;
+
+  static void check_exec_space(Kokkos::View<ValueType *, DeviceType>)
+  {
+    static_assert(std::is_same<Kokkos::Serial,
+                               typename DeviceType::execution_space>::value,
+                  "");
+  }
+
+  static void sort(Kokkos::View<ValueType *, DeviceType> view)
+  {
+    check_exec_space(view);
+
+    std::sort(view.data(), view.data() + view.extent(0));
+  }
+
+  static auto computePermutation(Kokkos::View<ValueType *, DeviceType> view)
+  {
+    check_exec_space(view);
+
+    using ViewType = Kokkos::View<ValueType *, DeviceType>;
+    using ExecutionSpace = typename ViewType::execution_space;
+
+    int const n = view.extent(0);
+
+    Kokkos::View<SizeType *, DeviceType> permute(
+        Kokkos::ViewAllocateWithoutInitializing("permute"), n);
+    iota(permute);
+
+    std::sort(permute.data(), permute.data() + n,
+              [&view](size_t const &a, size_t const &b) {
+                return view(a) < view(b);
+              });
+
+    return permute;
+  }
+
+  static Kokkos::View<SizeType *, DeviceType>
+  sortAndComputePermutation(Kokkos::View<ValueType *, DeviceType> view)
+  {
+    check_exec_space(view);
+
+    using ViewType = Kokkos::View<ValueType *, DeviceType>;
+    using ExecutionSpace = typename ViewType::execution_space;
+
+    int const n = view.extent(0);
+
+    auto permute = computePermutation(view);
+
+    Kokkos::View<int *, DeviceType> view_copy("view_copy", n);
+    Kokkos::deep_copy(view_copy, view);
+    Kokkos::parallel_for(
+        "apply_permutation", Kokkos::RangePolicy<ExecutionSpace>(0, n),
+        KOKKOS_LAMBDA(int i) { view(i) = view_copy(permute(i)); });
```

Comment:
So we generate the inverse permutation here?

Reply:
Good point. This seems to be an error.